h2o-py/tests/pyunit_utils/utils_for_glm_tests.py

```diff
@@ -0,0 +1,150 @@
+                             constraint_eta0=0.1258925, constraint_tau=10, constraint_alpha=0.1, 
+                             constraint_beta=0.9, constraint_c0=10):
+    """
+    This function given the parameters will return a constraint GLM model.
+    """
+    if linear_constraints is None:
+        raise H2OValueError("linear_constraints cannot be None")
+        
+    params = {"family":family, "lambda_":0.0, "seed":12345, "remove_collinear_columns":True, "solver":solver, 
+              "linear_constraints":linear_constraints, "init_optimal_glm":init_optimal_glm, 
+              "constraint_eta0":constraint_eta0, "constraint_tau":constraint_tau, "constraint_alpha":constraint_alpha,
+              "constraint_beta":constraint_beta, "constraint_c0":constraint_c0}
+    if beta_constraints is not None:
+        params['beta_constraints']=beta_constraints
+        params["separate_linear_beta"]=separate_linear_beta
+    if startval is not None:
+        params["startval"]=startval
+        
+    constraint_glm = glm(**params)
+    constraint_glm.train(x=x, y=y, training_frame=training_dataset)
+    return constraint_glm
+
+def constraint_glm_gridsearch(training_dataset, x, y, solver="AUTO", family="gaussia", linear_constraints=None,
+                              beta_constraints=None, metric="logloss", return_best=True, startval=None, 
+                              init_optimal_glm=False, constraint_eta0=[0.1258925],  constraint_tau=[10], 
+                              constraint_alpha=[0.1], constraint_beta=[0.9], constraint_c0=[10], epsilon=1e-3):
+    """
+    This function given the obj_eps_hyper and inner_loop_hyper will build and run a gridsearch model and return the one
+    with the best metric.
+    """
+    if linear_constraints is None:
+        raise H2OValueError("linear_constraints cannot be None")
+
+    params = {"family":family, "lambda_":0.0, "seed":12345, "remove_collinear_columns":True, "solver":solver,
+              "linear_constraints":linear_constraints}
+    hyper_parameters = {"constraint_eta0":constraint_eta0, "constraint_tau":constraint_tau, "constraint_alpha":constraint_alpha,
+                   "constraint_beta":constraint_beta, "constraint_c0":constraint_c0}
+    if beta_constraints is not None:
+        params['beta_constraints']=beta_constraints
+        hyper_parameters["separate_linear_beta"] = [True, False]
+    if startval is not None:
+        params["startval"]=startval
+    if init_optimal_glm:
+        params["init_optimal_glm"]=True
+        
+    glmGrid = H2OGridSearch(glm(**params), hyper_params=hyper_parameters)
+    glmGrid.train(x=x, y=y, training_frame=training_dataset)
+    sortedGrid = glmGrid.get_grid()
+    print(sortedGrid)
+    if return_best:
+        print_model_hyperparameters(sortedGrid.models[0], hyper_parameters)
+        return sortedGrid.models[0]
+    else:
+        return grid_models_analysis(sortedGrid.models, hyper_parameters, metric=metric, epsilon=epsilon)
+
```

Comment:
```suggestion


```
Ideally, I would try to stick to PEP-8 and [keep 2 blank lines between top-level functions](https://peps.python.org/pep-0008/#blank-lines).